pphysx3d/src/shapes/mod.rs

```diff
@@ -23,7 +26,62 @@ struct Ray {
 pub struct GameObject {
     pub shape: Box<dyn Shape>,    // The collider
     pub position: Isometry3<f32>, // includes a translation vector and a rotation part as an unit quaternion
-    pub velocity: Vector3<f32>,
+    velocity: Vector3<f32>,
+    acceleration: Vector3<f32>,
+    force_accum: Vector3<f32>,
     //texture:
-    pub mass: f32,
+    pub inverse_mass: f32,
+    pub restitution: f32, // elasticity aka bounciness. rename to bounciness?
+    pub friction: f32,    // coefficient of friction
+}
+
+impl GameObject {
+    pub fn set_velocity(&mut self, velocity: Vector3<f32>) {
+        self.velocity = velocity;
+    }
+
+    pub fn get_velocity(&self) -> Vector3<f32> {
+        return self.velocity;
+    }
+
+    pub fn set_mass(&mut self, mass: f32) {
+        self.inverse_mass = 1. / mass;
+    }
+
+    pub fn set_inverse_mass(&mut self, inverse_mass: f32) {
+        self.inverse_mass = inverse_mass;
+    }
+
+    pub fn get_mass(&self) -> f32 {
```

Comment:
getters are also just mass(&self)

Reply:
I use get_mass and set_mass because I changed the field to inverse_mass. Almost every calculation involving mass requires dividing by it, and then an object with 0 mass (which shouldn't exist, but a user can set mass to zero) will be bad bad bad. Also, with inverse_mass = 0 we can set mass to infinity for immovable objects.

Reply:
Fair enough. I just meant that "get" isn't used in getters

Reply:
Ah, true true.